server/pulp/server/managers/content/orphan.py

```diff
@@ -236,10 +238,18 @@ def delete_orphan_content_units_by_type(type_id):
 
         :param type_id: id of the content type
         :type type_id: basestring
+        :param content_unit_ids: list of content unit ids to delete; None means delete them all
+        :type content_unit_ids: iterable or None
         """
         # get the model matching the type
         content_model = plugin_api.get_unit_model_by_id(type_id)
-        content_units = content_model.objects().only('id', '_storage_path')
+        if content_unit_ids:
+            query_sets = []
+            for page in plugin_misc.paginate(content_unit_ids):
+                query_sets.append(content_model.objects(id__in=page).only('id', '_storage_path'))
```

Comment:
I learned recently about mongoengine's caching behavior, and it's non-intuitive. Would you expect that each QuerySet instance keeps a copy of every result it yields? Me either. You'll need to call ``.no_cache()`` on each QuerySet instance to prevent that.